### PR TITLE
Fix shellcheck failures of hack/grab-profiles.sh

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix shellcheck failures of hack/grab-profiles.sh

**Which issue(s) this PR fixes**:

Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/grab-profiles.sh [hunk 1]
```diff
@@ -65,7 +65,7 @@ output_dir="."
 tunnel_port="${tunnel_port:-1234}"
 
 args=$(getopt -o s:mho:k:c -l server:,master,heapster,output:,kubelet:,scheduler,controller-manager,help,inuse-space,inuse-objects,alloc-space,alloc-objects,cpu,kubelet-binary:,master-binary:,scheduler-binary:,controller-manager-binary:,scheduler-port:,controller-manager-port: -- "$@")
-if [[ $? -ne 0 ]]; then
+if [[ $? ]]; then
```
Comment (fejta) [inline hack/grab-profiles.sh]: `if ! args=$(getopt -o ...); then` (does this work?)

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

File: hack/grab-profiles.sh (hunk 1 above)
Comment (SataQiu) [reply hack/grab-profiles.sh]: I think that should work too, but this already can work here.

**Pull request merged**